crates/fluvio-stream-model/src/store/dual_store.rs

```diff
@@ -474,7 +473,17 @@ mod listener {
             self.event_publisher().current_change()
         }
 
-        pub async fn listen(&mut self) {
+        /// returns once there is at least one change recorded by the the event_publisher
+        /// can only be called once so consumes the ChangeListener
+        pub async fn listen_for_at_least_one_change(mut self) {
+            if self.last_change != 0 {
+                error!("listen_for_at_least_one_change should only be called on a newly constructed ChangeListener");
```

Comment:
Can you return error instead of logging?

Reply:
Actually, we can enforce in the interface.